### PR TITLE
MVP: New phone number input

### PR DESCRIPTION
**Pull request by dominicaw**
**Why:**
Intl tel input library did not have the level of flexibility we needed to create a mobile first phone input field. We have opted to move away from Intl tel input, in favour of a custom built phone field.

**What:**
I have taken the [country codes object](https://github.com/jackocnr/intl-tel-input/blob/master/src/js/data.js) from Intl tel input library, removed unnecessary information and formatted it. [Select2](https://select2.org/) is using that information to create a dropdown with a search field.

**MVP Requirements**
1. Easy to search & select a country code on desktop and mobile
2. Basic phone number validation based on length
3. Auto selects appropriate country code based on override field in webflow symbol

**See it in action**
Check it out on [Webflow](https://offerzen.webflow.io/employers/test-new-mobile-input)

### CODE REVIEW

File: offerzen/global/employers/employers@7.js [hunk 1]
```diff
@@ -0,0 +1,1618 @@
...
+
+    function normaliseCharacters(term) {
+      return term
+        .toLocaleLowerCase()
+        .normalize('NFD') // Creates separate bytes for characters and diacritical marks
+        .replace(/[\u0300-\u036f]/g, ''); // Replaces any diacritical marks in the string
+    }
+
+    // Country Data Set
+    const phoneNumberCountries = [
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@7.js:130]: Please move this to the top of phone number input for clarity (just after the comment)

File: offerzen/global/employers/employers@7.js [hunk 2]
```diff
@@ -0,0 +1,1618 @@
...
+      .toLowerCase();
+
+    // Initialise Select2
+    countryCodeSelector.select2({
+      data: phoneNumberCountries,
+      templateResult: formatData,
+      templateSelection: formatCode,
+      matcher: matchCustom,
+      dropdownParent: $('.js-country-code-dropdown'),
+    });
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@7.js:1419]: Please move this into a function  - you can call it initializePhoneNumberField, or setupPhoneNumberField or similar

File: offerzen/global/employers/employers@7.js [hunk 3]
```diff
@@ -0,0 +1,1618 @@
...
+        if (!field) {
+          return;
+        }
+        field.focus();
+        $('.select2-search__field').attr(
+          'placeholder',
+          'Search by country or code'
+        );
+      });
+    });
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@7.js:1442]: Move this content into the above function, too, please

File: offerzen/global/employers/employers@7.js [hunk 4]
```diff
@@ -0,0 +1,1618 @@
...
+    phoneNumberInput.on('keyup input blur', updatePhoneNumber);
+
+    // Trigger validation
+    countryCodeSelector.on('blur', function () {
+      $(phoneNumberInput).parsley().validate();
+    });
+
+    phoneNumberInput.on('blur', function () {
+      $(phoneNumberInput).parsley().validate();
+    });
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@7.js:1488]: Also move this into the initialize function

File: offerzen/global/employers/employers@7.js [hunk 5]
```diff
@@ -0,0 +1,1618 @@
...
+        originalCompanyLeadFormLoaded();
+      }
+
+      function onPhoneFieldReady() {
+        isPhoneReady = true;
+        onFormReady();
+      }
+
+      updateSubscribeToHiringInsightsField();
+      addPhoneValidation(onPhoneFieldReady);
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@7.js:1607]: You can call your "initalize" function just before adding the validator

**Review by jethroflanagan — CHANGES_REQUESTED** [review]
These changes should make it clearer as to the flow of code. Otherwise what it's doing looks good!

**Review by chama-ram — CHANGES_REQUESTED** [review]
Looks good! 👌👌Please update the script for the Experiment 1 lead form as well.

**Review by jethroflanagan — APPROVED** [review]
Great stuff!

**Review by chama-ram — APPROVED** [review]
Yay! Thank you for updating the scripts 🥇Looking goood!

**Pull request merged**